src/txn/txn.c

```diff
@@ -1183,6 +1185,7 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
     if (fix_upd != NULL)
         WT_ERR(__txn_fixup_prepared_update(session, hs_cursor, fix_upd, commit));
 
+prepare_verify:
```

Comment:
If we are committing, we need to add the check for update state progressing to `WT_PREPARE_RESOLVED`

Reply:
Yes, instead of commit change, it can be verified with `WT_PREAPARE_RESOLVED`

Reply:
I guess something on lines of:

`WT_ASSERT(session, !commit || head_upd->txnid != txn->id || upd->prepapre_state == WT_PREPARE_RESOLVED)`

Reply:
Yes, fixed it instead of commit flag, checked the update txnid itself to be aborted.